components/script/dom/servoparser/mod.rs

```diff
@@ -813,6 +814,27 @@ impl ParserContext {
             pushed_entry_index: None,
         }
     }
+
+    pub(crate) fn append_parent_to_csp_list(&self, policy_container: Option<&PolicyContainer>) {
+        let Some(policy_container) = policy_container else {
+            return;
+        };
+        let Some(parent_csp_list) = &policy_container.csp_list else {
+            return;
+        };
+        let Some(parser) = self.parser.as_ref().map(|p| p.root()) else {
+            return;
+        };
+        let new_csp_list = match parser.document.get_csp_list() {
+            None => parent_csp_list.clone(),
+            Some(original_csp_list) => {
+                let mut appended_csp_list = original_csp_list.clone();
+                appended_csp_list.append(parent_csp_list.clone());
+                appended_csp_list.to_owned()
+            },
+        };
+        parser.document.set_csp_list(Some(new_csp_list));
```

Comment:
I also wonder how we should handle the csp_list on both document and global. Here, we set it on the document. But in many of the checks, we use the window global to retrieve the policy container. Are they shared in some way? Shouldn't we have only 1 CSP list per realm, thus the window and document one should point to the same thing?

Reply:
The global getter just fetches the container from the appropriate place according to the kind of global it is, so we get the same result if we obtain it from the global or the global's document.